### PR TITLE
Support nonpersistent config

### PR DESCRIPTION
**Pull request by liangyuanpeng**
#50 

Support config of non-persistent topic by default.

In my case,  some time i just use non-persistent for mop , i want my user use topic `topicA` not `non-persistent://public/default/topicA`.

### CODE REVIEW

**Comment by codelipenghui** [comment]
@liangyepianzhou Currently, if you want to use the non-persistent topic, you can just use the non-persistent as the topic name domain such as `non-persistent://public/default/a/b/c`, the MoP will use the non-persistent topic. Does this works for you?

**Comment by codelipenghui** [comment]
@liangyepianzhou Oh, sorry. I missed some context that you have mentioned in https://github.com/streamnative/mop/issues/50, the change is clear for me, but looks like there are some unrelated changes, could you please rebase/merge the master branch?

**Comment by liangyuanpeng** [comment]
>  could you please rebase/merge the master branch? 

Seems like have some problem about commit log message after rebase master, i will push again later.

**Comment by Technoboy-** [comment]
> > could you please rebase/merge the master branch?
> 
> Seems like have some problem about commit log message after rebase master, i will push again later.

All right. thanks.

File: mqtt-impl/src/main/java/io/streamnative/pulsar/handlers/mqtt/MQTTServerConfiguration.java [hunk 1]
```diff
@@ -97,6 +97,13 @@
     )
     private String defaultNamespace = "default";
 
+    @FieldContext(
+            category = CATEGORY_MQTT,
+            required = false,
+            doc = "Default Pulsar topic persistent that the MQTT server used."
+    )
+    private boolean defaultTopicPersistent = true;
```
Comment (codelipenghui) [inline mqtt-impl/src/main/java/io/streamnative/pulsar/handlers/mqtt/MQTTServerConfiguration.java:105]: It should be defaultTopicDomain = persistent, if users want to change to non-persistent, just change the defaultTopicDomain = non-persistent

**Comment by liangyuanpeng** [comment]
Tested:

1. subscribe topic `helloworld3`
2. send message to topic `helloworld3`
3. consumer received message
4. topic print log   
```
23:28:04.520 [pulsar-ordered-OrderedExecutor-7-0-EventThread] INFO  org.apache.pulsar.broker.service.BrokerService - Created topic NonPersistentTopic{topic=non-persistent://public/default/helloworld3}
```

**Comment by codelipenghui** [comment]
@liangyepianzhou Could you please help check the failed test? looks related to the check style, you can run ` mvn checkstyle:check && mvn license:check && mvn spotbugs:check` locally to check if there are checkstyle issues.

**Comment by Technoboy-** [comment]
@liangyepianzhou 
Install [checkstyle](https://stackoverflow.com/questions/26955766/intellij-idea-checkstyle) plugin. Add configuration file from source directory. Set checkstyle version to 8.32.
And scan the files that you modify to fix all the checkstyle issues.

File: mqtt-impl/src/main/java/io/streamnative/pulsar/handlers/mqtt/utils/PulsarTopicUtils.java [hunk 2]
```diff
@@ -121,13 +126,14 @@ public static String getPulsarTopicName(String mqttTopicName, String defaultTena
             return TopicName.get(domain, tenant, namespace,
                     urlEncoded ? URLEncoder.encode(localName) : localName).toString();
         } else {
-            return TopicName.get(TopicDomain.persistent.value(), defaultTenant, defaultNamespace,
+            //TopicDomain.persistent
```
Comment (Technoboy-) [inline mqtt-impl/src/main/java/io/streamnative/pulsar/handlers/mqtt/utils/PulsarTopicUtils.java:129]: Remove the comment

File: mqtt-impl/src/test/java/io/streamnative/pulsar/handlers/mqtt/untils/PulsarTopicUtilsTest.java [hunk 3]
```diff
@@ -56,10 +59,29 @@ public void testGetTopicFilter() {
...
         Pair<TopicDomain, NamespaceName> pair2 = PulsarTopicUtils.getTopicDomainAndNamespaceFromTopicFilter(
-                "persistent://public/default//a/b/c", "public", "default");
+                "persistent://public/default//a/b/c", "public", "default",TopicDomain.persistent.value());
         Assert.assertEquals(pair1, pair2);
 
     }
+
+    @Test
+    public void testasyncGetTopicListFromTopicSubscriptionForDefaultTopicDomain() {
+        //test default topic domaon for  non-persistent
```
Comment (Technoboy-) [inline mqtt-impl/src/test/java/io/streamnative/pulsar/handlers/mqtt/untils/PulsarTopicUtilsTest.java:71]: domaon -> domain.

File: mqtt-impl/src/test/java/io/streamnative/pulsar/handlers/mqtt/untils/PulsarTopicUtilsTest.java [hunk 4]
```diff
@@ -56,10 +59,29 @@ public void testGetTopicFilter() {
...
+        List<String> topics1 = PulsarTopicUtils.asyncGetTopicListFromTopicSubscription(
+                "/a/b/c", "public", "default", null, "non-persistent").join();
+        List<String> topics2 = PulsarTopicUtils.asyncGetTopicListFromTopicSubscription(
+                "non-persistent://public/default//a/b/c", "public", "default", null, "non-persistent").join();
+        Assert.assertEquals(topics1, topics2);
+
+        //test default topic domaon for persistent
```
Comment (Technoboy-) [inline mqtt-impl/src/test/java/io/streamnative/pulsar/handlers/mqtt/untils/PulsarTopicUtilsTest.java:78]: domaon -> domain.

File: mqtt-impl/src/main/java/io/streamnative/pulsar/handlers/mqtt/MQTTServerConfiguration.java [hunk 5]
```diff
@@ -97,6 +97,14 @@
     )
     private String defaultNamespace = "default";
 
+
+    @FieldContext(
+            category = CATEGORY_MQTT,
+            required = true,
+            doc = "Default Pulsar topic persistent that the MQTT server used."
```
Comment (Technoboy-) [inline mqtt-impl/src/main/java/io/streamnative/pulsar/handlers/mqtt/MQTTServerConfiguration.java:104]: Default Pulsar topic persistent -> Default Pulsar topic domain

File: mqtt-impl/src/main/java/io/streamnative/pulsar/handlers/mqtt/proxy/MQTTProxyConfiguration.java [hunk 6]
```diff
@@ -131,6 +131,13 @@
     )
     private String defaultNamespace = "default";
 
+    @FieldContext(
+            category = CATEGORY_MQTT,
+            required = true,
+            doc = "Default Pulsar topic persistent that the MQTT server used."
```
Comment (Technoboy-) [inline mqtt-impl/src/main/java/io/streamnative/pulsar/handlers/mqtt/proxy/MQTTProxyConfiguration.java:137]: Default Pulsar topic persistent -> Default Pulsar topic domain

**Review by Technoboy- — CHANGES_REQUESTED** [review]

File: mqtt-impl/src/main/java/io/streamnative/pulsar/handlers/mqtt/proxy/MQTTProxyConfiguration.java [hunk 7]
```diff
@@ -131,6 +131,13 @@
     )
     private String defaultNamespace = "default";
 
+    @FieldContext(
+            category = CATEGORY_MQTT,
+            required = true,
+            doc = "Default Pulsar topic that the MQTT server used."
```
Comment (Technoboy-) [inline mqtt-impl/src/main/java/io/streamnative/pulsar/handlers/mqtt/proxy/MQTTProxyConfiguration.java:137]: Default Pulsar topic that the MQTT server used -> Default Pulsar topic domain the MQTT server used.

File: mqtt-impl/src/main/java/io/streamnative/pulsar/handlers/mqtt/MQTTServerConfiguration.java [hunk 8]
```diff
@@ -97,6 +97,14 @@
     )
     private String defaultNamespace = "default";
 
+
+    @FieldContext(
+            category = CATEGORY_MQTT,
+            required = true,
+            doc = "Default Pulsar topic that the MQTT server used."
```
Comment (Technoboy-) [inline mqtt-impl/src/main/java/io/streamnative/pulsar/handlers/mqtt/MQTTServerConfiguration.java:104]: Default Pulsar topic that the MQTT server used. -> Default Pulsar topic domain the MQTT server used.

**Comment by liangyuanpeng** [comment]
```
Error:  Tests run: 17, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 77.72 s <<< FAILURE! - in io.streamnative.pulsar.handlers.mqtt.hivemq.HiveMQIntegrationTest
Error:  testBasicPublishAndConsumeWithMQTT(io.streamnative.pulsar.handlers.mqtt.hivemq.HiveMQIntegrationTest)  Time elapsed: 60.015 s  <<< FAILURE!
org.testng.internal.thread.ThreadTimeoutException: Method io.streamnative.pulsar.handlers.mqtt.hivemq.HiveMQIntegrationTest.testBasicPublishAndConsumeWithMQTT() didn't finish within the time-out 60000
	at com.hivemq.client.internal.mqtt.handler.publish.incoming.MqttIncomingPublishFlow.onCancel(MqttIncomingPublishFlow.java:183)
	at com.hivemq.client.internal.mqtt.handler.util.FlowWithEventLoop.cancel(FlowWithEventLoop.java:62)
	at com.hivemq.client.internal.mqtt.MqttBlockingClient$MqttPublishes.close(MqttBlockingClient.java:385)
	at com.hivemq.client.internal.mqtt.mqtt3.Mqtt3BlockingClientView$Mqtt3PublishesView.close(Mqtt3BlockingClientView.java:215)
	at io.streamnative.pulsar.handlers.mqtt.hivemq.HiveMQIntegrationTest.testBasicPublishAndConsumeWithMQTT(HiveMQIntegrationTest.java:57)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:124)
	at org.testng.internal.InvokeMethodRunnable.runOne(InvokeMethodRunnable.java:54)
	at org.testng.internal.InvokeMethodRunnable.run(InvokeMethodRunnable.java:44)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)

[INFO] Running io.streamnative.pulsar.handlers.mqtt.ProxyTest
[INFO] Tests run: 10, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 17.453 s - in io.streamnative.pulsar.handlers.mqtt.ProxyTest
[INFO] Running io.streamnative.pulsar.handlers.mqtt.SimpleIntegrationTest
[INFO] Tests run: 39, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 26.587 s - in io.streamnative.pulsar.handlers.mqtt.SimpleIntegrationTest
[INFO] Running io.streamnative.pulsar.handlers.mqtt.BasicAuthenticationIntegrationTest
[INFO] Tests run: 3, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 17.037 s - in io.streamnative.pulsar.handlers.mqtt.BasicAuthenticationIntegrationTest
[INFO] Running io.streamnative.pulsar.handlers.mqtt.ProxyTLSTest
[INFO] Tests run: 20, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 19.168 s - in io.streamnative.pulsar.handlers.mqtt.ProxyTLSTest
[INFO] Running io.streamnative.pulsar.handlers.mqtt.TokenAuthenticationIntegrationTest
[INFO] Tests run: 3, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 18.032 s - in io.streamnative.pulsar.handlers.mqtt.TokenAuthenticationIntegrationTest
[INFO] Running io.streamnative.pulsar.handlers.mqtt.TLSTest
[INFO] Tests run: 10, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 18.426 s - in io.streamnative.pulsar.handlers.mqtt.TLSTest
[INFO] 
[INFO] Results:
[INFO] 
Error:  Failures: 
Error:  io.streamnative.pulsar.handlers.mqtt.hivemq.HiveMQIntegrationTest.testBasicPublishAndConsumeWithMQTT(io.streamnative.pulsar.handlers.mqtt.hivemq.HiveMQIntegrationTest)
Error:    Run 1: HiveMQIntegrationTest.testBasicPublishAndConsumeWithMQTT:57 » ThreadTimeout Me...
[INFO]   Run 2: PASS
[INFO]   Run 3: PASS
[INFO]   Run 4: PASS
[INFO]   Run 5: PASS
[INFO]   Run 6: PASS
[INFO]   Run 7: PASS
[INFO] 
[INFO] 
Error:  Tests run: 96, Failures: 1, Errors: 0, Skipped: 0
```  

Seems like not about this PR.

**Comment by Technoboy-** [comment]
> ```
> Error:  Tests run: 17, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 77.72 s <<< FAILURE! - in io.streamnative.pulsar.handlers.mqtt.hivemq.HiveMQIntegrationTest
> Error:  testBasicPublishAndConsumeWithMQTT(io.streamnative.pulsar.handlers.mqtt.hivemq.HiveMQIntegrationTest)  Time elapsed: 60.015 s  <<< FAILURE!
> org.testng.internal.thread.ThreadTimeoutException: Method io.streamnative.pulsar.handlers.mqtt.hivemq.HiveMQIntegrationTest.testBasicPublishAndConsumeWithMQTT() didn't finish within the time-out 60000
> 	at com.hivemq.client.internal.mqtt.handler.publish.incoming.MqttIncomingPublishFlow.onCancel(MqttIncomingPublishFlow.java:183)
> 	at com.hivemq.client.internal.mqtt.handler.util.FlowWithEventLoop.cancel(FlowWithEventLoop.java:62)
> 	at com.hivemq.client.internal.mqtt.MqttBlockingClient$MqttPublishes.close(MqttBlockingClient.java:385)
> 	at com.hivemq.client.internal.mqtt.mqtt3.Mqtt3BlockingClientView$Mqtt3PublishesView.close(Mqtt3BlockingClientView.java:215)
> 	at io.streamnative.pulsar.handlers.mqtt.hivemq.HiveMQIntegrationTest.testBasicPublishAndConsumeWithMQTT(HiveMQIntegrationTest.java:57)
> 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
> 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
> 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
> 	at java.lang.reflect.Method.invoke(Method.java:498)
> 	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:124)
> 	at org.testng.internal.InvokeMethodRunnable.runOne(InvokeMethodRunnable.java:54)
> 	at org.testng.internal.InvokeMethodRunnable.run(InvokeMethodRunnable.java:44)
> 	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
> 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
> 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
> 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
> 	at java.lang.Thread.run(Thread.java:748)
> 
> [INFO] Running io.streamnative.pulsar.handlers.mqtt.ProxyTest
> [INFO] Tests run: 10, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 17.453 s - in io.streamnative.pulsar.handlers.mqtt.ProxyTest
> [INFO] Running io.streamnative.pulsar.handlers.mqtt.SimpleIntegrationTest
> [INFO] Tests run: 39, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 26.587 s - in io.streamnative.pulsar.handlers.mqtt.SimpleIntegrationTest
> [INFO] Running io.streamnative.pulsar.handlers.mqtt.BasicAuthenticationIntegrationTest
> [INFO] Tests run: 3, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 17.037 s - in io.streamnative.pulsar.handlers.mqtt.BasicAuthenticationIntegrationTest
> [INFO] Running io.streamnative.pulsar.handlers.mqtt.ProxyTLSTest
> [INFO] Tests run: 20, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 19.168 s - in io.streamnative.pulsar.handlers.mqtt.ProxyTLSTest
> [INFO] Running io.streamnative.pulsar.handlers.mqtt.TokenAuthenticationIntegrationTest
> [INFO] Tests run: 3, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 18.032 s - in io.streamnative.pulsar.handlers.mqtt.TokenAuthenticationIntegrationTest
> [INFO] Running io.streamnative.pulsar.handlers.mqtt.TLSTest
> [INFO] Tests run: 10, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 18.426 s - in io.streamnative.pulsar.handlers.mqtt.TLSTest
> [INFO] 
> [INFO] Results:
> [INFO] 
> Error:  Failures: 
> Error:  io.streamnative.pulsar.handlers.mqtt.hivemq.HiveMQIntegrationTest.testBasicPublishAndConsumeWithMQTT(io.streamnative.pulsar.handlers.mqtt.hivemq.HiveMQIntegrationTest)
> Error:    Run 1: HiveMQIntegrationTest.testBasicPublishAndConsumeWithMQTT:57 » ThreadTimeout Me...
> [INFO]   Run 2: PASS
> [INFO]   Run 3: PASS
> [INFO]   Run 4: PASS
> [INFO]   Run 5: PASS
> [INFO]   Run 6: PASS
> [INFO]   Run 7: PASS
> [INFO] 
> [INFO] 
> Error:  Tests run: 96, Failures: 1, Errors: 0, Skipped: 0
> ```
> 
> Seems like not about this PR.

Did you run successfully in your local ?

**Comment by liangyuanpeng** [comment]
> testBasicPublishAndConsumeWithMQTT

Yes.  sometime  timeout.  

testIntegration

**Review by Technoboy- — APPROVED** [review]
LGTM

**Review by gaoran10 — APPROVED** [review]
LGTM

**Pull request merged**